secure_test.go

```diff
@@ -1271,7 +1272,55 @@ func TestModifyResponseHeadersWithSSL(t *testing.T) {
 	err := s.ModifyResponseHeaders(res)
 	expect(t, err, nil)
 
-	expect(t, res.Header.Get("Location"), "https://example.com")
+	expect(t, res.Header.Get("Location"), "http://example.com")
+}
+
+func TestModifyResponseHeadersWithSSLAndMatchingSSLHost(t *testing.T) {
+	s := New(Options{
+		SSLRedirect:     true,
+		SSLHost:         "secure.example.com",
+		SSLProxyHeaders: map[string]string{"X-Forwarded-Proto": "https"},
+	})
+
+	req, _ := http.NewRequest("GET", "/foo", nil)
+	req.Host = "www.example.com"
+	req.URL.Scheme = "http"
+	req.Header.Add("X-Forwarded-Proto", "https")
+
+	res := &http.Response{}
+	res.Header = http.Header{"Location": []string{"http://secure.example.com"}}
+	res.Request = req
+
+	expect(t, res.Header.Get("Location"), "http://secure.example.com")
+
+	err := s.ModifyResponseHeaders(res)
+	expect(t, err, nil)
+
+	expect(t, res.Header.Get("Location"), "https://secure.example.com")
+}
+
+func TestModifyResponseHeadersWithSSLAndPortInLocationResponse(t *testing.T) {
+	s := New(Options{
+		SSLRedirect:     true,
+		SSLHost:         "secure.example.com",
+		SSLProxyHeaders: map[string]string{"X-Forwarded-Proto": "https"},
+	})
+
+	req, _ := http.NewRequest("GET", "/foo", nil)
+	req.Host = "www.example.com"
+	req.URL.Scheme = "http"
+	req.Header.Add("X-Forwarded-Proto", "https")
+
+	res := &http.Response{}
+	res.Header = http.Header{"Location": []string{"http://secure.example.com:877"}}
+	res.Request = req
+
+	expect(t, res.Header.Get("Location"), "http://secure.example.com:877")
+
+	err := s.ModifyResponseHeaders(res)
+	expect(t, err, nil)
+
+	expect(t, res.Header.Get("Location"), "http://secure.example.com:877")
```

Comment:
These tests look good to me but shouldn't there be some tests for locations that include a path, like `http://secure.example.com/path/page.html` ?